### ISSUE TITLE
TF get does not return a list of errors, rather it returns a single differing error upon invocation.

### ISSUE THREAD

**Issue by notjames** [hashicorp/terraform#10935]
### Terraform Version
0.8.2

### Affected Resource(s)
I'm using multiple aws resources.

### Terraform Configuration Files
https://drive.google.com/open?id=0B2DscNnnn26oUkN4eGhDaHJfUUE

### Debug Output
https://gist.github.com/notjames/572d4c5480ee986021a26846f7a1c309

Succinctly put:

```shell
$ while true;do terraform get -update=true | grep -v Get;sleep 1;done
Error loading Terraform: module instances.root: module site-vpn: required variable private-subnets not set
Error loading Terraform: module instances.root: module site-vpn: required variable private-subnets not set
Error loading Terraform: module instances.root: module site-vpn: required variable sg-external not set
Error loading Terraform: module instances.root: module site-vpn: required variable public-subnets not set
Error loading Terraform: module instances.root: module site-vpn: required variable environment not set
Error loading Terraform: module instances.root: module site-vpn: required variable azones not set
Error loading Terraform: module instances.root: module site-vpn: required variable environment not set
Error loading Terraform: module instances.root: module site-vpn: required variable public-subnets not set
Error loading Terraform: module instances.root: module site-vpn: required variable project-name not set
Error loading Terraform: module instances.root: module site-vpn: required variable private-subnets not set
Error loading Terraform: module instances.root: module site-vpn: required variable sg-external not set
Error loading Terraform: module instances.root: module site-vpn: required variable private_ip not set
Error loading Terraform: module instances.root: module site-vpn: required variable sg-external not set
Error loading Terraform: module instances.root: module site-vpn: required variable sg-external not set
Error loading Terraform: module instances.root: module site-vpn: required variable azones not set
Error loading Terraform: module instances.root: module site-vpn: required variable azones not set
Error loading Terraform: module instances.root: module site-vpn: required variable azones not set
Error loading Terraform: module instances.root: module site-vpn: required variable private-subnets not set
Error loading Terraform: module instances.root: module site-vpn: required variable sg-vpn not set
Error loading Terraform: module instances.root: module site-vpn: required variable environment not set
```

### Panic Output
N/A

### Expected Behavior
Given any /one/ or /set/ of issue(s) with HCL, TF should reliably give me one error per issue consistently per invocation IE if there's only one issue, then I should reliably get the same error on every invocation if that error is never fixed. If I have five errors then TF should give me the same five errors on every invocation if those errors are never fixed. If it's not possible for TF to give me a LIST of errors then I should at least get the same single error upon invocation until that error is fixed. 

### Actual Behavior
TF spits out a single different problem per invocation. Note that no changes to HCL were made between each invocation.

### Steps to Reproduce
1. terraform get

### Important Factoids
None of which I'm aware except that I'm a complete n00b to TF and I'm aware that the HCL is probably all kinds of wrong. That may be a contributing factor.

### References
N/A

**Comment by notjames** [comment]
Please note that I've updated this issue with output from an updated set of HCLs. It's my opinion, however, that you don't need the updated HCLs since the HCLs I've provided reliably reproduce the issue. However, if you feel you need updates HCLs, please let me know and I will attempt to provide.

**Comment by jbardin** [comment]
Hi @notjames,

Thanks for opening the issue. Yes, Terraform aborts at the first error it sees while validating the config, and I agree this can make cleaning up errors tedious.

Accumulating the validation errors should be fairly simple. Let me see about getting those to the UI in a sane manner.

**Comment by notjames** [comment]
Thank you @jbardin; much appreciated.

**Comment by jbardin** [comment]
closed via #12383
